src/index.ts

```diff
@@ -76,6 +77,10 @@ export interface Options {
    * Generate unknown type instead of any
    */
   unknownAny: boolean
+  /**
+   * Custom function to provide a type name for a given schema
+   */
+  customName?: (schema: LinkedJSONSchema, keyNameFromDefinition: string | undefined) => string
```

Comment:
A bit different from the proposal in https://github.com/bcherny/json-schema-to-typescript/issues/181#issuecomment-421885339, I think it would be more useful to expose the schema object directly, as both `id` and `title` comes from the schema object. This would allow demands like #456 to be easily achieved from the userland.